oneview/resource_connection_templates.go

```diff
@@ -102,7 +102,7 @@ func resourceConnectionTemplates() *schema.Resource {
 func resourceConnectionTemplatesRead(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
 	id := d.Id()
-	cTemplate, err := config.ovClient.GetConnectionTemplateByName(id)
+	cTemplate, err := config.ovClient.GetConnectionTemplateByURI(utils.Nstring(id))
 	if err != nil || cTemplate.URI.IsNil() {
 		d.SetId("")
 		return nil
```

Comment:
According to me resourceConnectionTemplatesUpdate have two issues:
1. it is trying to update all the properties(include in schema)...like state, status, modified, created..i think we should not raise put call for those
2. uri := strings.Split(updateOptions.URI.String(), "/")[3]
	template, err := config.ovClient.UpdateConnectionTemplate(uri, updateOptions)
  why we need to do split operation then getting the id and pass it to it? we can just pass the uri itself

Reply:
done